data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/partition/MongoDBExportPartitionSupplier.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.mongo.export.partition;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import org.bson.Document;
+import org.opensearch.dataprepper.model.source.coordinator.PartitionIdentifier;
+import org.opensearch.dataprepper.plugins.mongo.client.MongoDBConnection;
+import org.opensearch.dataprepper.plugins.mongo.client.BsonHelper;
+import org.opensearch.dataprepper.plugins.mongo.coordination.partition.ExportPartition;
+import org.opensearch.dataprepper.plugins.mongo.configuration.MongoDBSourceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class MongoDBExportPartitionSupplier implements Function<ExportPartition, List<PartitionIdentifier>> {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoDBExportPartitionSupplier.class);
+    private static final String MONGODB_PARTITION_KEY_FORMAT = "%s|%s|%s|%s"; // partition format: <db.collection>|<gte>|<lt>|<className>
+    private static final String COLLECTION_SPLITTER = "\\.";
+
+    private final MongoDBSourceConfig sourceConfig;
+
+    public MongoDBExportPartitionSupplier(final MongoDBSourceConfig sourceConfig) {
+        this.sourceConfig = sourceConfig;
+    }
+
+    private List<PartitionIdentifier> buildPartitions(final ExportPartition exportPartition) {
+        final List<PartitionIdentifier> collectionPartitions = new ArrayList<>();
+        final String collectionDbName = exportPartition.getCollection();
+        List<String> collection = List.of(collectionDbName.split(COLLECTION_SPLITTER));
+        if (collection.size() < 2) {
+            throw new IllegalArgumentException("Invalid Collection Name. Must be in db.collection format");
+        }
+        try (MongoClient mongoClient = MongoDBConnection.getMongoClient(sourceConfig)) {
+            final MongoDatabase db = mongoClient.getDatabase(collection.get(0));
+            final MongoCollection<Document> col = db.getCollection(collection.get(1));
+            final int partitionSize = exportPartition.getPartitionSize();
+            FindIterable<Document> startIterable = col.find()
+                    .projection(new Document("_id", 1))
+                    .sort(new Document("_id", 1))
+                    .limit(1);
+            while (true) {
```

Comment:
Placing loops that run like this deep in the call stack can make debugging very difficult. We should move the loop into the `Runnable` method.

Reply:
Added `!Thread.currentThread().isInterrupted()` for while loop and moved it to same package. We need the partition supplier to be injectable from DocDB source (next PR), so didn't make it package protected.